test-integration/controllers/MetricsDBSpec.scala

```diff
@@ -0,0 +1,81 @@
+package controllers
+
+import com.gu.editorialproductionmetricsmodels.models.{MetricOpt, ProductionOffice}
+import database.MetricsDB
+import helpers.{PostgresHelpers, TestData}
+import models.ProductionMetricsError
+import models.db.{ForkResponse, Metric, MetricsFilters}
+import org.joda.time.DateTime
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{FunSuite, Matchers}
+import play.api.mvc.Results
+
+import scala.util.Right
+
+class MetricsDBSpec extends FunSuite with MockitoSugar with Matchers with Results with PostgresHelpers {
+  implicit val metricsDb: MetricsDB = new MetricsDB()
+
+  private[this] def metricToMetricOpt(metric: Metric): MetricOpt = MetricOpt(
+    id = Some(metric.id),
+    originatingSystem = Some(metric.originatingSystem),
+    composerId = metric.composerId,
+    storyBundleId = metric.storyBundleId,
+    commissioningDesk = metric.commissioningDesk,
+    userDesk = metric.userDesk,
+    inWorkflow = Some(metric.inWorkflow),
+    inNewspaper = Some(metric.inNewspaper),
+    creationTime = Some(metric.creationTime),
+    firstPublicationTime = metric.firstPublicationTime,
+    roundTrip = Some(metric.roundTrip),
+    productionOffice = metric.productionOffice,
+    issueDate = metric.issueDate,
+    bookSectionName = metric.bookSectionName,
+    bookSectionCode = metric.bookSectionCode,
+    newspaperBook = metric.newspaperBook,
+    newspaperBookSection = metric.newspaperBookSection
+  )
+
+  private[this] val testMetric: Metric = TestData.randomMetricWith("cId_upsert", "sbId_upsert")
+
+  test("Upsert metric - insert metric") {
+    val result: Either[ProductionMetricsError, Metric] = metricsDb.updateOrInsert(None, metricToMetricOpt(testMetric))
+    result shouldBe a [Right[_,_]]
+
+    val actualMetric = result.right.get
+    val expectedMetric = testMetric.copy(id = actualMetric.id)
+    actualMetric shouldBe expectedMetric
+  }
+
+  test("Upsert metric - update metric") {
```

Comment:
Having the same description for multiple tests makes it hard to work which one failed